src/Microsoft.DotNet.Darc/DarcLib/Helpers/DependencyFileManager.cs

```diff
@@ -121,24 +123,37 @@ public async Task<JObject> ReadDotNetToolsConfigJsonAsync(string repoUri, string
         }
     }
 
+
+    /// <summary>
+    /// Get the tools.dotnet section of the global.json from a target repo URI
+    /// </summary>
+    /// <param name="repoUri">repo to get the version from</param>
+    /// <param name="commit">commit sha to query</param>
+    public async Task<SemanticVersion> ReadToolsDotnetVersionAsync(string repoUri, string commit)
+    {
+        JObject globalJson = await ReadGlobalJsonAsync(repoUri, commit);
+        JToken dotnet = globalJson.SelectToken("tools.dotnet", true);
+
+        _logger.LogInformation("Reading dotnet version from global.json succeeded!");
+
+        if (!SemanticVersion.TryParse(dotnet.ToString(), out SemanticVersion dotnetVersion))
+        {
+            _logger.LogError($"Failed to parse dotnet version from global.json from repo: {repoUri} at commit {commit}. Version: {dotnet}");
+        }
+
+        return dotnetVersion;
+    }
+
     public async Task<XmlDocument> ReadNugetConfigAsync(string repoUri, string branch)
     {
         return await ReadXmlFileAsync(VersionFiles.NugetConfig, repoUri, branch);
     }
 
     public async Task<VersionDetails> ParseVersionDetailsXmlAsync(string repoUri, string branch, bool includePinned = true)
     {
-        if (!string.IsNullOrEmpty(branch))
-        {
-            _logger.LogInformation(
-                $"Getting a collection of dependencies from '{VersionFiles.VersionDetailsXml}' in repo '{repoUri}' " +
-                $"and branch '{branch}'...");
-        }
-        else
-        {
-            _logger.LogInformation(
-                $"Getting a collection of dependencies from '{VersionFiles.VersionDetailsXml}' in repo '{repoUri}'...");
-        }
+        _logger.LogInformation(
```

Comment:
The PCS service will eventually run these methods, shouldn't we use structured logging?

Reply:
Maestro is already running this. You're right, we should, but the whole of DarcLib is not like that at the moment. Only the new code.

Reply:
I agree we can be updating it one statement at a time though.